tests/lib/rules/prefer-read-only-props.js

```diff
@@ -167,7 +167,7 @@ ruleTester.run('prefer-read-only-props', rule, {
         import React from "react";
 
         interface Props {
-          name: string;
+          readonly name: string;
```

Comment:
Without `readonly` this would be a failing case for TypeScript, unless I've misunderstood this test?

Reply:
then can we keep the test for "not TS", and add a new one with `readonly` for TS?

Reply:
This test already had the flag `ts`, it was passing before because the TypeScript rule had not been written.

I don't think this test was correct originally

Reply:
aha, that makes sense. maybe then we need a failing case without `readonly`? (or did you already add one)

Reply:
I've already done that for both interfaces and types 😄 

https://github.com/jsx-eslint/eslint-plugin-react/pull/3593/files#diff-a3687af456dbe6329995ef962d1467d5af3a22e7207d226e60b7e3fb03ed1969R441-R502